featuretools/synthesis/encode_features.py

```diff
@@ -159,11 +166,10 @@ def encode_features(feature_matrix, features, top_n=DEFAULT_TOP_N, include_unkno
                                       desc="Encoding pass 2",
                                       unit="feature")
     for c in iterator:
-        if c in extra_columns:
-            continue
-        try:
-            new_X[c] = pd.to_numeric(new_X[c], errors='raise')
-        except (TypeError, ValueError):
-            pass
+        if c in encoded_columns:
+            try:
+                new_X[c] = pd.to_numeric(new_X[c], errors='raise')
+            except (TypeError, ValueError):
+                pass
```

Comment:
If we can't figure out why this is here/if it's still necessary, maybe we should add a warning so that we can at least tell when it does end up throwing this error instead of quietly passing

Reply:
If we do add a warning, I think it'd be good to also test the warning to increase code coverage.

Reply:
Our current tests don't cover this except block - we would need to find an example where this could occur

Reply:
In the interest of getting this fix into today's release, I'm open to doing a separate issue / PR cycle on the usefulness of this block / how we want to handle the exceptions.